tests/testthat/test-cal-binary.R

```diff
@@ -0,0 +1,99 @@
+test_that("Binary breaks functions work", {
+  x10 <- .cal_binary_table_breaks(segment_logistic, Class, .pred_good)
+
+  expect_equal(sd(x10$predicted_midpoint), 0.3063903, tolerance = 0.000001)
```

Comment:
Locally, on my intel iMac, I get a lot of errors that look to be related to tolerance. I've set the tolerance as low as 0.01 and still get them though. 

```
ℹ Loading probably
[ FAIL 5 | WARN 0 | SKIP 0 | PASS 20 ]

── Failure (test-cal-binary.R:18): Binary breaks functions work ────────────────
sd(x12$predicted_midpoint) (`actual`) not equal to 0.2982188 (`expected`).

  `actual`: 0.293
`expected`: 0.298

── Failure (test-cal-binary.R:43): Binary logistic functions work ──────────────
sd(x22$prob) (`actual`) not equal to 0.2784706 (`expected`).

  `actual`: 0.27
`expected`: 0.28

── Failure (test-cal-binary.R:44): Binary logistic functions work ──────────────
mean(x22$prob) (`actual`) not equal to 0.4765672 (`expected`).

  `actual`: 0.50
`expected`: 0.48

── Failure (test-cal-binary.R:80): Binary windowed functions work ──────────────
sd(x32$predicted_midpoint) (`actual`) not equal to 0.3571706 (`expected`).

  `actual`: 0.31
`expected`: 0.36

── Failure (test-cal-binary.R:81): Binary windowed functions work ──────────────
mean(x32$predicted_midpoint) (`actual`) not equal to 0.4374119 (`expected`).

  `actual`: 0.41
`expected`: 0.44
```